libraries/botbuilder-dialogs-adaptive/src/languageGeneratorExtensions.ts

```diff
@@ -13,53 +13,42 @@ import { LanguageGenerator } from './languageGenerator';
 import { resourceExplorerKey } from './resourceExtensions';
 import { LanguagePolicy } from './languagePolicy';
 
-/**
- * redeclare module to inject functions
- */
-declare module 'botbuilder-dialogs/lib/dialogManager' {
-    export interface DialogManager {
-        useLanguageGeneration(lg?: string | LanguageGenerator): DialogManager;
-        useLanguagePolicy(policy: LanguagePolicy): DialogManager;
-    }
-}
-
 export const languageGeneratorKey = Symbol('LanguageGenerator');
 export const languageGeneratorManagerKey = Symbol('LanguageGeneratorManager');
+export const languagePolicyKey = Symbol('LanguagePolicy');
 
-/**
- * Register default LG file or a language generator as default language generator.
- */
-DialogManager.prototype.useLanguageGeneration = function(lg?: string | LanguageGenerator): DialogManager {
-    const _self = this as DialogManager;
-    const resourceExplorer: ResourceExplorer = _self.initialTurnState.get(resourceExplorerKey) || new ResourceExplorer();
-    let defaultLg = 'main.lg';
-    if (lg) {
-        if (typeof lg === 'string') {
-            defaultLg = lg;
+export class LanguageGeneratorExtensions {
+    /**
+     * Register default LG file or a language generator as default language generator.
+     */
+    public static useLanguageGeneration(dialogManager: DialogManager, lg?: string | LanguageGenerator): DialogManager {
+        const resourceExplorer: ResourceExplorer = dialogManager.initialTurnState.get(resourceExplorerKey) || new ResourceExplorer();
+        let defaultLg = 'main.lg';
+        if (lg) {
+            if (typeof lg === 'string') {
+                defaultLg = lg;
+            }
```

Comment:
Simplify to

```suggestion
        const defaultLg = lg && typeof lg === 'string' ? lg : 'main.lg';
```

Unfortunately both the `lg` truthy and `typeof` checks are required since `typeof` doesn't evaluate if the string is truthy....

Reply:
Got it, thanks for your suggestion.